src/test/java/seedu/address/testutil/DirectoryInitUtil.java

```
@@ -0,0 +1,57 @@
+package seedu.address.testutil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+
+/**
+ * A set of helper methods to manage temporary directories.
+ */
+public class DirectoryInitUtil {
+
+    /**
+     * Initialise a temporary directory by copying all files from {@code sourceDir} to {@code tempDir}.
+     */
+    public static void initializeTemporaryDirectory(Path sourceDir, Path tempDir) throws IOException {
+        Files
+            .walk(sourceDir)
+            .forEach(filePath -> copyFilesToTempDir(filePath, sourceDir, tempDir));
+    }
+
+    /**
+     * Helper method to copy a file located at {@code filePath} to {@code tempDir}. Files copied maintain
+     * relative directory structure with reference to {@code sourceDir}.
+     */
+    private static void copyFilesToTempDir(Path filePath, Path sourceDir, Path tempDir) {
+        Path relativePath = relativizeToTempDir(filePath, sourceDir, tempDir);
+        try {
+            Files.copy(filePath, relativePath, StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
```

we should throw the exception up instead (as is the case of createTemporaryFileInFolder).

---

If I throw the exception up, the lambda function in initializeTemporaryDirectory will not compile because the consumer lambda `forEach(filePath -> copyFilesToTempDir(filePath, sourceDir, tempDir));` cannot throw checked exceptions like IOException.
One way I can get around this is by rethrowing an AssertionError instead. I've seen our code do it before (e.g. in AddressBookSystemTest#assertSelectedCardChanged), but I'm not sure if this is a good practice.